### ISSUE TITLE
Full project restructure

### ISSUE THREAD

**Issue by prasannavl** [osm-in/flood-map#17]
Great job on quickly hacking up so far. But I guess, this has grown reasonably and so I thought I'll migrate it to a project structure with `npm`, and a build system - so we can slowly start modularizing it. 

Unfortunately, there is no `master` branch on this - so, I haven't created a `Pull Request` yet. If this seems okay to you guys, please create a master branch and let me know so I can create a PR, or jus t directly base it off my master here: https://github.com/prasannavl/flood-map/tree/master

I've also written `Travis` build scripts to auto create a travis build and push to the `gh-pages` branch whenever someone pushes to the `master` so it auto build using `gulp`. Does all the image minimization, compression, minification, runs tests if required, and auto deploys from `master`.

Will modify README accordingly if this makes sense to you guys. For testing it out, please clone my `master` branch, and run `npm install`, and the `gulp build` to get the local build. 

PS: Also for Travis an autodeply `ssh` key is required, which will have to be added after encrypting it to the `.travis` for auto commits to `gh-pages` - but all the base work is already done.

**Comment by planemad** [comment]
@prasannavl thank you for doing this! The rains are back and I just reset the data for new reports.

Are your changes still relevant?

**Comment by prasannavl** [comment]
I don't see why not. However, two things I'd take into account:
 
1. My branch is still a little old, since I see a few commits from last year here that wouldn't be there.
 Merging them may be as easy as it's just works, or might require a bit of manual work. 
2. Once 1 is done, everything including node, npm, and the deps have to be brought up to date. 

If you could do 1, I should be able to make some time towards the end of the week.
